coverage/docs/design.md

```diff
@@ -0,0 +1,165 @@
+# Overview
+This code coverage tool calculates per file, per package and overall coverage on target directories. It generates the following artifacts
+  - code coverage profile, which is produced by "[go test -coverprofile](https://blog.golang.org/cover)" and contains all block level code coverage data
+  - XML file that stores file-level and package-level code coverage, formatted to be readable by TestGrid
```

Comment:
Is this jUnit? Should call that out if it's not TestGrid specific

Reply:
@michelle192837 Are all junit XML readable by TestGrid? Does TestGrid have any specific requirement?

Reply:
There's some specific expectations for TestGrid to parse it; they're specified in Gubernator's section on GCS Layout, https://github.com/kubernetes/test-infra/tree/master/gubernator#gcs-layout

Reply:
That's layout in GCS, the actual formatting of the file is simple jUnit XML then